src/Parser/Core/Modules/Items/Legion/EyeOfCommand.js

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import ITEMS from 'common/ITEMS';
+import { calculateSecondaryStatDefault } from 'common/stats';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import { formatPercentage } from 'common/format';
+
+/**
+ * Eye of Command
+ * Your melee auto attacks increase your Critical Strike by X for 10 sec, stacking up to 10 times. This effect is reset if you auto attack a different target.
+ */
+class EyeOfCommand extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+  eyeOfCommand = {
+    [SPELLS.LEGIONS_GAZE.id]: {
```

Comment:
What do you need the individual start/uptimes for?

Reply:
It was my original idea of how to calculate average critGain by getting the uptime of each stack level and summing in the end and then it worked and was accurate.

Reply:
Aha, in that case I recommend keeping a total instead like here:
https://github.com/WoWAnalyzer/WoWAnalyzer/blob/bef2745c470bc4d1c5993d121731be76ce5ddb7b/src/Parser/Core/Modules/Items/Legion/AntorusTheBurningThrone/GarothiFeedbackConduit.js#L46
With that you can use a single getter to calculate the average:
https://github.com/WoWAnalyzer/WoWAnalyzer/blob/bef2745c470bc4d1c5993d121731be76ce5ddb7b/src/Parser/Core/Modules/Items/Legion/AntorusTheBurningThrone/GarothiFeedbackConduit.js#L22-L24

I believe this to be much simpler code, this will reduce the amount of code you need to calculate the average and be easier to track.

Reply:
is changebuffstack something we've fabricated?

Reply:
Yes, it was created to normalize buff stack handling.